fossunited/api/chapter.py

```diff
@@ -0,0 +1,38 @@
+import frappe
+
+from fossunited.doctype_ids import USER_PROFILE
+
+
+@frappe.whitelist()
+def check_if_chapter_member(
```

Comment:
We are checking with this on the frontend - where access to visibly see another chapter dashboard is blocked. 

How about the backend side? With read and write permissions enabled for a `Chapter Team Member`, with the use of APIs: 

1. Can a member of chapter member of A with roles assigned, assign himself as member of chapter B. 

2. By this also gain access to events of chapter B ?

Reply:
Would a check `before every save` help to validate if the edit is made by the member of the chapter ? This might potentially block the access via API methods ?

Reply:
![Screenshot 2024-08-10 at 1 43 52 AM](https://github.com/user-attachments/assets/3010580d-c2b0-4c40-b39c-dc16d2c22920)
The permissions set for `All` needs to be changed too @harshtandiya ?

Reply:
All shouldn't be a problem, as the chapter's info is public anyway

Reply:
> We are checking with this on the frontend - where access to visibly see another chapter dashboard is blocked. 
> 
> How about the backend side? With read and write permissions enabled for a `Chapter Team Member`, with the use of APIs: 
> 
> 1. Can a member of chapter member of A with roles assigned, assign himself as member of chapter B.
> 
> 2. By this also gain access to events of chapter B ?

Can we reference this in a new issue? This might bring additional errors. Won't be taking this up in this PR

Reply:
> All shouldn't be a problem, as the chapter's info is public anyway

There are chapter members' info linked to it as well - I think we had the same discussion when we were deciding permissions for Localhosts - and decided to remove `All`?